### ISSUE TITLE
Rendering HTML Tables with nested lists parses incorrectly

### ISSUE THREAD

**Issue by michaelryancaputo** [probablyup/markdown-to-jsx#170]
Hi there!

I have come upon a use case where we need to render tables, with lists nested within them. I have tested on prior versions of the component (v6.4.1) and the issue is not present. It seems to have started since 6.5.0.

This is an example using `6.4.1` - https://stackblitz.com/edit/react-qugqd6
This is an example using `6.6.1` - https://stackblitz.com/edit/react-6nelfr

You'll notice when using 6.6.1, it renders trailing `</li>` tags for some reason.

![screen shot 2018-04-02 at 9 10 37 am](https://user-images.githubusercontent.com/80901/38203922-bbd04cc0-3655-11e8-9467-d0c5b07ef053.png)

This is the HTML i'm using
```
<table border="1">
  <tr>
    <td>a</td>
    <td>b</td>
    <td>c</td>
  </tr>
  <tr>
    <td>left</td>
    <td>
      <p>Start of table</p>
      <ul>
        <li>List 1</li>
        <li>
          <ul>
            <li>Nested List 1</li>
          </ul>
        </li>
        <li>
          <ul>
            <li>list 2</li>
          </ul>
        </li>
      </ul>
    </td>
    <td>right</td>
  </tr>
</table>
```

**Comment by stephtr** [comment]
I think it's the same issue as #168.

**Comment by michaelryancaputo** [comment]
@stephtr Yup, looks like it. Thanks for doing that!

**Comment by stephtr** [comment]
But since the other issue is closed, we could keep this one opened.

**Comment by asumaran** [comment]
Any updates on this? it looks like this is the reason styleguidist is broken for me. Markdown is being incorrectly rendered.

**Comment by michaelryancaputo** [comment]
Bueller?

**Comment by probablyup** [comment]
I'll take another look this weekend.

**Comment by probablyup** [comment]
Released as 6.6.2
